layers/+lang/typescript/packages.el

```diff
@@ -103,8 +103,12 @@
   (when typescript-fmt-on-save
     (add-hook 'typescript-tsx-mode-hook 'spacemacs/typescript-fmt-before-save-hook))
   (spacemacs/set-leader-keys-for-major-mode 'typescript-tsx-mode
-    "="  'spacemacs/typescript-format
-    "p" 'spacemacs/typescript-open-region-in-playground))
+    "p" 'spacemacs/typescript-open-region-in-playground)
+  (pcase typescript-backend
```

Comment:
Directly accessing the typescript-backend is dangerous, you need to call the respective getter otherwise your code may not know how to defer the correct backend if non is explicitly set.

Example:
```elisp
(defun typescript/post-init-flycheck ()
  (spacemacs/enable-flycheck 'typescript-mode)
  (spacemacs/enable-flycheck 'typescript-tsx-mode)
  (cond ((eq (spacemacs//typescript-backend) 'tide)
         (typescript/set-tide-linter))
        ((eq (spacemacs//typescript-backend) 'lsp)
         (typescript/set-lsp-linter)))
```

Reply:
can do. I just saw that here: https://github.com/syl20bnr/spacemacs/blob/ac3c3546a2eaf5d05c27fa3527263360146ffd94/layers/%2Bframeworks/vue/funcs.el#L16
so monkey see, monkey do 😄

Reply:
so, you mean accessing typescript-backend falls over if it's set to nil? 
does pcase also work when using spacemacs//typescript-backend?

I didn't test the nil case 😅 but I assumed that would work and match the wildcard _
What do you mean by deferring the correct backend?

But come to think of it, the nil case is a bit more complicated than I first thought anyway, I'll actually have to test wether the lsp layer is used in that case

Reply:
https://github.com/syl20bnr/spacemacs/blob/a2ac1814c6f6c59f377d9c29e7ab87869f87eb37/layers/%2Blang/typescript/funcs.el#L15

ah ok now I see what you mean I think.
so, just to make sure I understood correctly - the getter will give me the correct backend if the typescript-backend variable is set to nil?

Reply:
ok I needed quite a bit of trial and error until I realized that `spacemacs//typescript-backend` needs to be quoted at that point :see_no_evil: I guess I should read the elisp manual

Reply:
:) jup that would help. 

Fun fact while elisp is a poor choice for modern age programs clojure is pretty up-to-date in all application areas, so investing some time into learning lisp is not a bad thing, I would just start with clojure cause elisp is becoming more similar to it with every new release.